CenterForOpenScience/osf.io
# [IN-285] Add withdrawal requests for non-moderated providers to admin

## Purpose

Allow osf admins to handle preprint withdrawal requests from non-moderated
providers.

## Changes

- Add `approve`, `reject`, `list` views and templates for preprint requests
- Add migration to add preprint request perms to admin group
- Add tests

## QA Notes
- Prep: Need to have some withdrawal requests generated: given there is no UI, someone will staging-2 shell access needs to do this.

- On main (left) side menu: go to Preprints and click on `withdrawal requests`
- You should see a list of withdrawal requests, please make sure all field show. and their status is `pending`
- Try: `accept` and `reject` actions and see the status of selected requests changes (upon refresh).
- Make sure that the requests that just got accepted/rejected can't be selected again.

- Click on the preprint detail button of a pending requests: and the withdrawal request should show at the bottom of the page with both `reject` and `approve` buttons. Reject/Approve and make sure no errors and the status gets updated. Make sure that after rejecting or accepting (after automatic refresh) both buttons disappear. 

## Documentation


## After
<img width="1417" alt="screen shot 2018-06-29 at 6 06 15 pm" src="https://user-images.githubusercontent.com/4511563/42537207-30fe7ed6-8462-11e8-8575-2467ba856c48.png">

<img width="1402" alt="screen shot 2018-06-29 at 6 05 44 pm" src="https://user-images.githubusercontent.com/4511563/42537215-385eb038-8462-11e8-9d16-4febb15e2ff4.png">

<img width="1400" alt="screen shot 2018-06-29 at 5 42 59 pm" src="https://user-images.githubusercontent.com/4511563/42537230-469d1162-8462-11e8-9319-8ba998e9b225.png">

## Ticket

[IN-285](https://openscience.atlassian.net/browse/IN-285)
